src/Utilities/StringUtility.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * A class of utilities for dealing with namespaces.
+ *
+ * @package Automattic\WooCommerce\Utilities
+ */
+
+namespace Automattic\WooCommerce\Utilities;
+
+/**
+ * A class of utilities for dealing with namespaces.
+ */
+final class StringUtility {
```

Comment:
What would you think about naming this class just `String`? The code would read much better, and the class is behind a namespace so that being a too generic name isn't a problem.

Reply:
Also just `Utils` instead of `Utilities` as the namespace? (not sure about this one as I'm not a native English speaker, rather I'm an English-abuser)

Reply:
`String` is a reserved word but I've made it `Str`, let me know if that works!

As for the `Utilities` I generally prefer the longer form as a namespace, what do you think?